velox/functions/iceberg/tests/BucketFunctionTest.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/iceberg/BucketFunction.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/iceberg/tests/IcebergFunctionBaseTest.h"
+
+#include <gtest/gtest.h>
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::functions::iceberg;
+
+namespace {
+
+class BucketFunctionTest
+    : public functions::iceberg::test::IcebergFunctionBaseTest {
+ public:
+  BucketFunctionTest() {
+    options_.parseDecimalAsDouble = false;
+  }
+
+ protected:
+  template <typename T>
+  std::optional<int32_t> bucket(
+      std::optional<int32_t> numBuckets,
+      std::optional<T> value) {
+    return evaluateOnce<int32_t>("bucket(c0, c1)", numBuckets, value);
+  }
+
+  void assertBucket(
```

Comment:
bucket is a simple function; what's the reason for testing it using vectors?

Reply:
I use it to test decimal and date before, but I find evaluateOnce also can test it. I have updated it. Thanks for your reminder.